### PR TITLE
[CLOCK] Modified date & middle digits displays

### PR DESCRIPTION
**Pull request by JackZ974**
[CLOCK] modified date format & display of seconds/year

Fixes #[issue_no]

**Summary of Changes**

    Added the extraction of Seconds and Year from the Time string in order to display it with a smaller font as IRL
    The display of date is now MON/DAY/YEAR as expected
    removed the display of ":" in the date display

**Screenshots (if necessary)**

Before:
image
After:
image
Date display:
image
Real one:
image

**Additional context**

Discord username:

### CODE REVIEW

**Comment by camilo625** [comment]
Works as described. I am seeing CHR, UTC and ET in the textures (really blurry), not shure if that's normal.

**Comment by JackZ974** [comment]
> 
> 
> Works as described. I am seeing CHR, UTC and ET in the textures (really blurry), not shure if that's normal.

The texture is not mine, but considering the size of the overall texture file, not so much details can be added, I'm afraid. AFAIK, even the small blue legends below the white one were unreadable and couldn't make it to the texture

**Comment by veikkos** [comment]
I'm confused. Why doesn't the "before" image have the CHR/UTC/ET texts like there currently is with master? Have you removed them? But not in this PR?

**Comment by JackZ974** [comment]
Simply because, the rate of PR is going so fast in this project that I gave up on constanatly updating my sim when I'm coding. So i still have the old texture without the texts

**Review by AdenFlorian — APPROVED** [review]
ready to test! (it needs a retest because the code changed since last test)

**Comment by JackZ974** [comment]
Minor addition in the css of a dark colored background that lights up at night. 
image

**Comment by starvingjavi** [comment]
Tested during the day and the brightness of the numbers is not uniform between them like the reference pictures. During the night the brightness difference is not so apparent but it is still there
Captura de pantalla 2020-09-15 201813
Captura de pantalla 2020-09-15 201838

File: A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/Clock/A320_Neo_Clock.css [hunk 1]
```diff
@@ -51,34 +51,55 @@ a320-neo-clock-element {
   overflow: hidden; }
   a320-neo-clock-element #Mainframe {
     position: absolute;
+    /* top: -4%; */
```
Comment (Benjozork) [inline A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/Clock/A320_Neo_Clock.css:54]: Please remove comments ;)

File: A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/Clock/A320_Neo_Clock.css [hunk 2]
```diff
@@ -51,34 +51,55 @@ a320-neo-clock-element {
...
     bottom: 0;
     left: 2%;
-    right: 0; }
+    right: 0; 
+    background-color: rgba(5, 40, 80, 0.623);
+    color: rgba(155, 148, 148, 0.911);}
     a320-neo-clock-element #Mainframe #TopSelector {
       margin-top: 0%;
+      /* width: 100%;
+      height: 35%; */
```
Comment (Benjozork) [inline A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/Clock/A320_Neo_Clock.css:64]: Please remove comments ;)

File: A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/Clock/A320_Neo_Clock.css [hunk 3]
```diff
@@ -51,34 +51,55 @@ a320-neo-clock-element {
...
       a320-neo-clock-element #Mainframe #MiddleSelector #MiddleSelectorValue {
         font-family: Alarm-Clock;
-        font-size: 65px;
-        text-align: center;
+        font-size: 81px;
+        margin-left: 2%;
+        margin-right: 5%;
+        width: 95%;
+        text-align: right;
+        /* font-kerning: none; */
```
Comment (Benjozork) [inline A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/Clock/A320_Neo_Clock.css:91]: Please remove comments ;)

File: A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/Clock/A320_Neo_Clock.js [hunk 4]
```diff
@@ -57,10 +60,25 @@ class A320_Neo_Clock extends BaseAirliners {
             }
 
             if (this.middleSelectorElem) {
-                const currentDisplayTime = SimVar.GetSimVarValue("L:PUSH_CHRONO_SET", "Bool") ? this.getUTCDate() : this.getUTCTime()
+                let currentDisplayTime = "";
+                let currentDisplayTime2 = "";
+                if (SimVar.GetSimVarValue("L:PUSH_CHRONO_SET", "Bool") == true) {
```
Comment (Benjozork) [inline A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/Clock/A320_Neo_Clock.js:65]:
```suggestion
                if (SimVar.GetSimVarValue("L:PUSH_CHRONO_SET", "Bool") === 1) {
```

`== true`

File: A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/Clock/A320_Neo_Clock.js [hunk 5]
```diff
@@ -57,10 +60,25 @@ class A320_Neo_Clock extends BaseAirliners {
...
+
```
Comment (Benjozork) [inline A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/Clock/A320_Neo_Clock.js:66]:
```suggestion
```

File: A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/Clock/A320_Neo_Clock.js [hunk 6]
```diff
@@ -57,10 +60,25 @@ class A320_Neo_Clock extends BaseAirliners {
...
+                    currentDisplayTime = this.getUTCDate(); 
+                    currentDisplayTime2 = this.getUTCYear(); 
+
```
Comment (Benjozork) [inline A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/Clock/A320_Neo_Clock.js:69]:
```suggestion
```

File: A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/Clock/A320_Neo_Clock.js [hunk 7]
```diff
@@ -57,10 +60,25 @@ class A320_Neo_Clock extends BaseAirliners {
...
+                    currentDisplayTime2 = this.getUTCYear(); 
+                    
+                } else {
+                    currentDisplayTime = this.getUTCTime(); 
+                    currentDisplayTime2 = this.getUTCSeconds(); 
+                }
                 if (currentDisplayTime !== this.lastDisplayTime) {
                     this.lastDisplayTime = currentDisplayTime;
                     this.middleSelectorElem.textContent = currentDisplayTime;
+
```
Comment (Benjozork) [inline A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/Clock/A320_Neo_Clock.js:77]:
```suggestion
```

File: A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/Clock/A320_Neo_Clock.js [hunk 8]
```diff
@@ -78,25 +96,36 @@ class A320_Neo_Clock extends BaseAirliners {
...
+    }
     getUTCDate() {
-        const Day = SimVar.GetGlobalVarValue("ZULU DAY OF MONTH", "number")
-        const Month = SimVar.GetGlobalVarValue("ZULU MONTH OF YEAR", "number")
-        const Year = `${SimVar.GetGlobalVarValue("ZULU YEAR", "number")}`.substr(2,4)
-
-        return `${Day}.${Month}.${Year}`
+        const Day = SimVar.GetGlobalVarValue("ZULU DAY OF MONTH", "number");
+        const Month = `${SimVar.GetGlobalVarValue("ZULU MONTH OF YEAR", "number")}`.padStart(2,'0');
+        return `${Month}.${Day}`;
```
Comment (Benjozork) [inline A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/Clock/A320_Neo_Clock.js:116]:
```suggestion
        const day = SimVar.GetGlobalVarValue("ZULU DAY OF MONTH", "number");
        const month = `${SimVar.GetGlobalVarValue("ZULU MONTH OF YEAR", "number")}`.padStart(2,'0');
        return `${month}.${day}`;
```

File: A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/Clock/A320_Neo_Clock.js [hunk 9]
```diff
@@ -78,25 +96,36 @@ class A320_Neo_Clock extends BaseAirliners {
...
     }
 
+    getUTCYear() {
+        const Year = `${SimVar.GetGlobalVarValue("ZULU YEAR", "number")}`.substr(2,4);
+        return `${Year}`;
```
Comment (Benjozork) [inline A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/Clock/A320_Neo_Clock.js:121]:
```suggestion
        const year = `${SimVar.GetGlobalVarValue("ZULU YEAR", "number")}`.substr(2,4);
        return `${year}`;
```

File: A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/Clock/A320_Neo_Clock.js [hunk 10]
```diff
@@ -6,15 +6,18 @@ class A320_Neo_Clock extends BaseAirliners {
         this.lastChronoState = null;
         this.lastChronoTime = null;
         this.lastDisplayTime = null;
+        this.lastDisplayTime2 = null;
         this.lastFlightTime = null;
         this.lastLocalTime = 0;
         this.lastResetVal = null;
+
```
Comment (Benjozork) [inline A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/Clock/A320_Neo_Clock.js:13]:
```suggestion
```

**Review by Benjozork — COMMENTED** [review]
Thank you for your PR, and nice use of `const` everywhere possible :)

Few things to correct but looking good ! I put suggestions on each so you can quickly fix it :)

**Review by UncleClapton — CHANGES_REQUESTED** [review]

File: A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/Clock/A320_Neo_Clock.css (hunk 2 above)
Comment (JackZ974) [reply A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/Clock/A320_Neo_Clock.css]: Done

File: A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/Clock/A320_Neo_Clock.js (hunk 5 above)
Comment (JackZ974) [reply A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/Clock/A320_Neo_Clock.js:66]: This was from a previous PR, not my part of code, but OK

File: A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/Clock/A320_Neo_Clock.js (hunk 8 above)
Comment (JackZ974) [reply A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/Clock/A320_Neo_Clock.js:114]: Was like this in the original ASOBO code

File: A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/Clock/A320_Neo_Clock.js (hunk 9 above)
Comment (JackZ974) [reply A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/Clock/A320_Neo_Clock.js:119]: this var was labelled like this in the original ASOBO code

File: A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/Clock/A320_Neo_Clock.js [hunk 11]
```diff
@@ -78,25 +96,36 @@ class A320_Neo_Clock extends BaseAirliners {
...
 
+    getUTCSeconds() {
+        const value = SimVar.GetGlobalVarValue("ZULU TIME", "seconds");
+        if (value) {
+            const seconds = Number.parseInt(value);
+            const time = Utils.SecondsToDisplayTime(seconds, true, true, false);
+            return time.toString().substr(6,2);
+        }
+        return "";
+    }
```
[the remark this answers is not in the document]
Comment (JackZ974) [reply A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/Clock/A320_Neo_Clock.js:109]: Good point. I'll look into that

File: A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/Clock/A320_Neo_Clock.js (hunk 9 above)
[the remark this answers is not in the document]
Comment (JackZ974) [reply A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/Clock/A320_Neo_Clock.js:118]: Ok. not so familiar with the subtleties of js coding. Always interesting to learn from the js Masters

**Comment by JackZ974** [comment]
> 
> 
> Tested during the day and the brightness of the numbers is not uniform between them like the reference pictures. During the night the brightness difference is not so apparent but it is still there
> Captura de pantalla 2020-09-15 201813
> Captura de pantalla 2020-09-15 201838
> 
> Tried changing the brightness knobs to see if it would change anything and nothing, it stays the same
You probably had the early version of the css
i modified the css in the meantime for the background and corrected this issue as well. Would you mind testing it again? This should be ok now

File: A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/Clock/A320_Neo_Clock.js (hunk 11 above)
[the remark this answers is not in the document]
Comment (JackZ974) [reply A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/Clock/A320_Neo_Clock.js]: Done the modification. Thx for pointing this out

File: A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/Clock/A320_Neo_Clock.css (hunk 3 above)
Comment (JackZ974) [reply A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/Clock/A320_Neo_Clock.css]: I will leave this one for now as a placeholder. This waiting for a new monospaced font to be creted for the chrono. #fonts guys are on it as it seems

File: A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/Clock/A320_Neo_Clock.js (hunk 4 above)
Comment (JackZ974) [reply A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/Clock/A320_Neo_Clock.js]: triple === ?

File: A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/Clock/A320_Neo_Clock.css (hunk 3 above)
Comment (Benjozork) [reply A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/Clock/A320_Neo_Clock.css]: > I will leave this one for now as a placeholder. This waiting for a new monospaced font to be created for the chrono. #fonts guys are on it as it seems
> 

Okay perfect :)

**Comment by JackZ974** [comment]
image

**Review by AdenFlorian — APPROVED** [review]
ready to test again!

**Review by UncleClapton — APPROVED** [review]
LGTM!

**Comment by ImenesFBW** [comment]
Works and looks as described. It's really sleek and adds a lot to cockpit immersion. Also you can see the min markers clearly from the pilot seat now.  Can't wait to see this in the master thanks for your work!

Now this might be unrelated but will add it here for future reference. Had to click the cabin ALL a few times for ecam to register it off the C/L.

**Pull request merged**